HEN_HOUSE/egs++/geometry/egs_lattice/egs_lattice.h

```diff
@@ -0,0 +1,920 @@
+/*
+###############################################################################
+#
+#  EGSnrc egs++ lattice geometry
+#  Copyright (C) 2019 Rowan Thomson and Martin Martinov
+#
+#  This file is part of EGSnrc.
+#
+#  EGSnrc is free software: you can redistribute it and/or modify it under
+#  the terms of the GNU Affero General Public License as published by the Free
+#  Software Foundation, either version 3 of the License, or (at your option)
+#  any later version.
+#
+#  EGSnrc is distributed in the hope that it will be useful, but WITHOUT
+#  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or 
+#  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License
+#  for more details.
+#
+#  To see the GNU Affero General Public License at:
+# <http://www.gnu.org/licenses/>.
+#
+################################################################################
+#
+#  When egs_lattice is used for publications, please cite the following paper:
+#  
+#  Martinov, Martin P., and Rowan M. Thomson.
+#  "Taking EGSnrc to new lows: Development of egs++ lattice geometry
+#  and testing with microscopic geometries."
+#  Medical Physics (2020).
+#
+###############################################################################
+#
+#  Author:          Martin Martinov, 2019
+#
+###############################################################################
+*/
+
+
+/*! \file egs_lattice.cpp
+ *  \brief A Bravais, cubic, and hexagonal lattice geometry
+
+  \ingroup Geometry
+
+A geometry which embeds a lattice of one geometry (named subgeometry below)
+into one region of a second geometry (named base geometry).  This geometry
+effectively recurses the subgeometry at every position defined by a Bravais,
+cubic, or hexagonal lattice.  As such, you can model an infinite amount of
+subgeometries (e.g., region 0 of egs_space) and the only slow down to your
+simulation would depend on how many subgeometries you would expect over a
+particle track.
+As this geometry only stores a single subgeometry in memory, it can only
+score in ALL subgeometries (or rather, the single subgeometry at all lattice
+positions) at once.  Therefore, dose to the subgeometry at different
+locations cannot be discerned.
```

Comment:
I would also add the paper reference here. Users a likely to be confused as to how this geometry differs from a simple repeater geometry.